src/validator.cpp

```diff
@@ -1635,6 +1708,530 @@ void Validator::ValidatorImpl::validateMathMLElements(const XmlNodePtr &node, co
+        addMathMLIssue("Math has a '" + node->name() + "' element without at least one MathML sibling.",
+                       Issue::ReferenceRule::MATH_MATHML,
+                       component);
+
+        return false;
+    }
+
+    return true;
+}
+
+bool Validator::ValidatorImpl::hasTwoMathmlSiblings(const XmlNodePtr &parentNode,
+                                                    const XmlNodePtr &node,
+                                                    const ComponentPtr &component)
+{
+    if (mathmlChildCount(parentNode) != 3) {
+        addMathMLIssue("Math has a '" + node->name() + "' element without two MathML siblings.",
+                       Issue::ReferenceRule::MATH_MATHML,
+                       component);
+
+        return false;
+    }
+
+    return true;
+}
+
+bool Validator::ValidatorImpl::hasAtLeastTwoMathmlSiblings(const XmlNodePtr &parentNode,
+                                                           const XmlNodePtr &node,
+                                                           const ComponentPtr &component)
+{
+    if (mathmlChildCount(parentNode) < 3) {
+        addMathMLIssue("Math has a '" + node->name() + "' element without at least two MathML siblings.",
+                       Issue::ReferenceRule::MATH_MATHML,
+                       component);
+
+        return false;
+    }
+
+    return true;
+}
+
+size_t Validator::ValidatorImpl::hasOneOrTwoMathmlSiblings(const XmlNodePtr &parentNode,
+                                                           const XmlNodePtr &node,
+                                                           const ComponentPtr &component)
+{
+    auto childCount = mathmlChildCount(parentNode);
+
+    if ((childCount != 2) && (childCount != 3)) {
+        addMathMLIssue("Math has a '" + node->name() + "' element without one or two MathML siblings.",
+                       Issue::ReferenceRule::MATH_MATHML,
+                       component);
+
+        return 0;
+    }
+
+    return childCount - 1;
+}
+
+bool Validator::ValidatorImpl::isFirstMathmlChild(const XmlNodePtr &parentNode,
+                                                  const XmlNodePtr &node,
+                                                  const ComponentPtr &component)
+{
+    if (!mathmlChildNode(parentNode, 0)->equals(node)) {
+        addMathMLIssue("Math has a '" + node->name() + "' element which is not the first MathML sibling.",
+                       Issue::ReferenceRule::MATH_MATHML,
+                       component);
+
+        return false;
+    }
+
+    return true;
+}
+
+bool Validator::ValidatorImpl::isSecondMathmlChild(const XmlNodePtr &parentNode,
+                                                   const XmlNodePtr &node,
+                                                   const ComponentPtr &component)
+{
+    if (!mathmlChildNode(parentNode, 1)->equals(node)) {
+        addMathMLIssue("Math has a '" + node->name() + "' element which is not the second MathML sibling.",
```

Comment:
child not sibling

Reply:
Nope, wrong name of method in fact.

46c115db